internal/models/filter.go

```diff
@@ -132,51 +132,74 @@ func (f *Filter) Apply(
 	return true
 }
 
+// IsActive checking if filter is active
 func (f *Filter) IsActive() bool {
 	return f.isActive
 }
 
 // shouldNamePass returns true if the metric should pass, false if should drop
 // based on the drop/pass filter parameters
 func (f *Filter) shouldNamePass(key string) bool {
-	if f.namePass != nil {
+
+	pass := func(f *Filter) bool {
 		if f.namePass.Match(key) {
 			return true
 		}
 		return false
 	}
 
-	if f.nameDrop != nil {
+	drop := func(f *Filter) bool {
 		if f.nameDrop.Match(key) {
 			return false
 		}
+		return true
 	}
+
+	if f.namePass != nil && f.nameDrop != nil {
+		return pass(f) && drop(f)
```

Comment:
I know it's an old PR, but whenever I look at this code (because the docs aren't specific enough) I find this sequence confusing. The definition of `drop(f)` means "the drop filter says 'pass'", which inverts the prosaic sense of the code. `if pass and drop` just doesn't smell good. Would it be clearer to remove the negation inside `drop` , and use these?
```
  return pass(f) && ! drop(f)
  return pass(f)
  return ! drop(f)
```